src/Dictionaries/HashedArrayDictionary.cpp

```diff
@@ -406,14 +406,19 @@ void HashedArrayDictionary<dictionary_key_type>::createAttributes()
 template <DictionaryKeyType dictionary_key_type>
 void HashedArrayDictionary<dictionary_key_type>::updateData()
 {
-    if (!update_field_loaded_block || update_field_loaded_block->rows() == 0)
+    if (!update_field_loaded_block)
     {
         QueryPipeline pipeline(source_ptr->loadUpdatedAll());
-
         PullingPipelineExecutor executor(pipeline);
         Block block;
+
         while (executor.pull(block))
         {
+            if (!block.rows())
+                continue;
+
+            convertToFullIfSparse(block);
```

Comment:
I added `convertToFullIfSparse()` here for consistency with `mergeBlockWithPipe()` below and with other dictionaries too.